tools/targets/REALTEK_RTL8195AM.py

```diff
@@ -1,141 +1,272 @@
+    write_fixed_width_value(get_version_by_time(), 16, output)
+    write_fixed_width_string(RAM2_SHA, 64, output)
+    write_fixed_width_value(RAM2_RSVD, 8, output)
+
+    append_image_file(image, output)
+    output.close()
+
+    ota = open(image_ota, "wb")
+    write_fixed_width_value(os.stat(image).st_size, 8, ota)
+    write_fixed_width_value(int(entry), 8, ota)
+    write_fixed_width_value(int(segment), 8, ota)
+    write_fixed_width_value(0xFFFFFFFF, 8, ota)
+    write_fixed_width_value(get_version_by_time(), 16, ota)
+    write_fixed_width_string(RAM2_SHA, 64, ota)
+    write_fixed_width_value(RAM2_RSVD, 8, ota)
+
+    append_image_file(image, ota)
+    ota.close()
+
+def find_symbol(toolchain, mapfile, symbol):
+    ret = None
+
+    HEX = '0x0{,8}(?P<addr>[0-9A-Fa-f]{8})'
+    if toolchain == "GCC_ARM":
+        SYM = re.compile(r'^\s+' + HEX + r'\s+' + symbol + '\r?$')
+    elif toolchain in ["ARM_STD", "ARM", "ARM_MICRO"]:
+        SYM = re.compile(r'^\s+' + HEX + r'\s+0x[0-9A-Fa-f]{8}\s+Code.*\s+i\.' + symbol + r'\s+.*$')
+    elif toolchain == "IAR":
+        SYM = re.compile(r'^' + symbol + r'\s+' + HEX + '\s+.*$')
+
+    with open(mapfile, 'r') as infile:
+        for line in infile:
+            match = re.match(SYM, line)
+            if match:
+                ret = match.group("addr")
+
+    if not ret:
+        print "[ERROR] cannot find the address of symbol " + symbol
+        return 0
+
+    return int(ret,16) | 1
+
+def parse_load_segment_gcc(image_elf):
+    # Program Headers:
+    #   Type           Offset   VirtAddr   PhysAddr   FileSiz MemSiz  Flg Align
+    #   LOAD           0x000034 0x10006000 0x10006000 0x026bc 0x026bc RW  0x8
+    #   LOAD           0x0026f0 0x30000000 0x30000000 0x06338 0x06338 RWE 0x4
+    segment_list = []
+    cmd = 'arm-none-eabi-readelf -l ' + image_elf
+    for line in subprocess.check_output(cmd, shell=True, universal_newlines=True).split("\n"):
+        if not line.startswith("  LOAD"):
+            continue
+        segment = line.split()
+        if len(segment) != 8:
+            continue
+        offset = int(segment[1][2:], 16)
+        addr   = int(segment[2][2:], 16)
+        size   = int(segment[4][2:], 16)
+        if addr != 0 and size != 0:
+            segment_list.append((offset, addr, size))
+    return segment_list
+
+def parse_load_segment_armcc(image_elf):
+    # ====================================
+    #
+    # ** Program header #2
+    #
+    #     Type          : PT_LOAD (1)
+    #     File Offset   : 52 (0x34)
+    #     Virtual Addr  : 0x30000000
+    #     Physical Addr : 0x30000000
+    #     Size in file  : 27260 bytes (0x6a7c)
+    #     Size in memory: 42168 bytes (0xa4b8)
+    #     Flags         : PF_X + PF_W + PF_R + PF_ARM_ENTRY (0x80000007)
+    #     Alignment     : 8
+    #
+    (offset, addr, size) = (0, 0, 0)
+    segment_list = []
+    in_segment = False
+    cmd = 'fromelf.exe --text -v --only=none ' + image_elf
```

Comment:
@Archcady has submitted a patch to drop .exe extension. This issue should have been fixed.